SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -0,0 +1,112 @@
+//===-- logger.hpp - Declarate functions and containers with that will be used
+//      for run tests with different types. -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the function that let iterate over
+/// provided types and provide each type to the call call-operator from the
+/// provided class/struct.
+///
+//===----------------------------------------------------------------------===//
+
+namespace esimd_test {
+namespace api {
+namespace functional {
+namespace type_coverage {
+
+// Type pack to store types and underlying data type names to use with
+// type_name_string
+template <typename... T> struct named_type_pack {
+  const std::string names[sizeof...(T)];
+
+  template <typename... nameListT>
+  named_type_pack(nameListT &&... nameList)
+      : names{std::forward<nameListT>(nameList)...} {}
+};
+
+enum class tested_types { all_types, fp_types, uint_types, sint_types };
+
+// Constructing named_type_pack with types that depent on chosen field of
+// tested_types enumeration
+template <tested_types T> auto get_tested_types() {
```

Comment:
```suggestion
template <tested_types required> auto get_tested_types() {
```
To use `T` for enum is a bit misleading

Reply:
this comment was applied in this commit:  334063fc311a13b4bb841d02ad12da7cd0ab26e3